### PR TITLE
Factory sets: Add more EV validation

### PR DESCRIPTION
**Pull request by MacChaeger**

### CODE REVIEW

File: data/mods/gen7/bss-factory-sets.json [hunk 1]
```diff
@@ -160,6 +160,7 @@
 			"species": "Aron",
 			"item": "Berry Juice",
 			"ability": "Sturdy",
+			"evs": {},
```
Comment (KrisXV) [inline data/mods/gen7/bss-factory-sets.json:163]: Why? This is intentional so you don’t have empty objects

File: test/sim/misc/random-teams.js [hunk 2]
```diff
@@ -181,11 +182,17 @@ describe(`Factory sets`, function () {
...
 
+						assert(!!set.evs, `Set of ${species} has no EVs specified`);
+						const keys = Object.keys(set.evs);
+						const evKeys = ['hp', 'atk', 'def', 'spa', 'spd', 'spe'];
 						let totalEVs = 0;
-						for (const ev in set.evs) {
+						for (const ev of keys) {
+							assert(evKeys.includes(ev), `Invalid EV key (${ev}) on set of ${species}`);
 							totalEVs += set.evs[ev];
 							assert((set.evs[ev] / 4) === Math.floor(set.evs[ev] / 4), `EVs of ${ev} not divisible by 4 on ${species}`);
```
Comment (AnnikaCodes) [inline test/sim/misc/random-teams.js:192]:
```suggestion
							assert.equal(set.evs[ev] % 4, 0, `EVs of ${ev} not divisible by 4 on ${species}`);
```

File: data/mods/gen7/bss-factory-sets.json (hunk 1 above)
Comment (MacChaeger) [reply data/mods/gen7/bss-factory-sets.json:163]: I think it's better to have one empty object for one set than to allow sets that don't have EVs by accident.

File: test/sim/misc/random-teams.js [hunk 3]
```diff
@@ -181,11 +182,17 @@ describe(`Factory sets`, function () {
...
+						const evKeys = ['hp', 'atk', 'def', 'spa', 'spd', 'spe'];
 						let totalEVs = 0;
-						for (const ev in set.evs) {
+						for (const ev of keys) {
+							assert(evKeys.includes(ev), `Invalid EV key (${ev}) on set of ${species}`);
 							totalEVs += set.evs[ev];
-							assert((set.evs[ev] / 4) === Math.floor(set.evs[ev] / 4), `EVs of ${ev} not divisible by 4 on ${species}`);
+							assert.equal(set.evs[ev] % 4, 0, `EVs of ${ev} not divisible by 4 on ${species}`);
 						}
+						const sortedKeys = Utils.sortBy([...keys], ev => evKeys.indexOf(ev));
```
Comment (Zarel) [inline test/sim/misc/random-teams.js:194]: Oh hey, you ended up using the thing I recommended; nice!

Comment (Zarel) [reply test/sim/misc/random-teams.js:194]: I would probably use `evOrder` rather than `evKeys` to be clearer about what it's for, though.

**Pull request merged**